subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/PmdPlugin.java

```diff
@@ -192,13 +193,21 @@ private void configureToolchains(Pmd task) {
         });
     }
 
-    private String calculateDefaultDependencyNotation(VersionNumber toolVersion) {
+    static Set<String> calculateDefaultDependencyNotation(final String versionString) {
+        final VersionNumber toolVersion = VersionNumber.parse(versionString);
         if (toolVersion.compareTo(VersionNumber.version(5)) < 0) {
-            return "pmd:pmd:" + extension.getToolVersion();
+            return Set.of("pmd:pmd:" + versionString);
         } else if (toolVersion.compareTo(VersionNumber.parse("5.2.0")) < 0) {
-            return "net.sourceforge.pmd:pmd:" + extension.getToolVersion();
+            return Set.of("net.sourceforge.pmd:pmd:" + versionString);
+        } else if (toolVersion.compareTo(VersionNumber.version(7)) >= 0) {
+            // starting from version 7, PMD is split into multiple modules
+            return Set.of(
```

Comment:
We need to support Java 8. `Set.of(...)` will not work

Reply:
no probs, I will migrate this

Reply:
should be JRE8 compatible now.